src/components/ChatInput/ChatInput.js

```diff
@@ -117,71 +105,60 @@ const ChatInput = ({ GOOGLE_CLIENT_ID }) => {
 
   useEffect(() => {
     if (editMessage.msg) {
-      setMessage(editMessage.msg);
+      messageRef.current.value = editMessage.msg;
     }
   }, [editMessage]);
 
   return (
-    <Box className={styles.container} border="2px solid #ddd">
-      {isUserAuthenticated && (
-        <Popup
+    <Box m="x20" border="2px solid #ddd">
+      <Box className={styles.container}>
+        <textarea
+          placeholder={isUserAuthenticated ? 'Message' : 'Sign in to chat'}
           disabled={!isUserAuthenticated}
-          trigger={
+          className={styles.textInput}
+          onChange={(e) => {
+            messageRef.current.value = e.target.value;
+            setDisableButton(!messageRef.current.value.length);
+            e.target.style.height = 'auto';
```

Comment:
can you please explain this part

Reply:
In this part, whenever text contents increases then the box height, the box height also increases upto 140px after that it becomes constant. Its similar to how the height varies in RC server.